R/utils.R

```diff
@@ -497,3 +497,31 @@ generate_css_dist_html <- function(url) {
     stop(sprintf("Invalid URL supplied in external_stylesheets. Please check the syntax used for this parameter."), call. = FALSE)
 }
 
+encode_plotly <- function(layout_objs) {
+  if (is.list(layout_objs) &&
+      "x" %in% names(layout_objs) &&
+      "visdat" %in% names(layout_objs$x)) {
+    # check to determine whether the current element is an
+    # object output from the plot_ly function; if it is,
+    # we can safely assume that it contains no other plot_ly
+    # objects and return the updated element as a mutated
+    # plotly figure argument that contains only data and
+    # layout attributes. we suppress messages since the
+    # plotly_build function will supply them, as it's typically
+    # run interactively.
+    obj <- suppressMessages(plotly::plotly_build(layout_objs)$x)
+    layout_objs <- obj[c("data", "layout")]
+    return(layout_objs)
+  } else {
```

Comment:
Is there anything we can bail out on here, and not fall into `seq_along` at all? Like perhaps pop the `is.list` up above out to wrap everything else:
```R
if(is.list(layout_objs)) {
  if("x" %in% names(layout_objs) && "visdat" %in% names(layout_objs$x)) {
    ...
  }
  else {
    for (i in seq_along(layout_objs)) {
      ...
    }
  }
}
layout_objs
```
I'm thinking particularly about simple values like strings making it here from callbacks.

Reply:
Thanks for the feedback, this feels like a much cleaner, leaner implementation now. 🤞that this is ready to :dancer:

Reply:
(hopefully) fixed in f1cc83f9b9c3122ea513299b348f0717d7142551